src/web/helpers/device/camera-data-backup.ts

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable no-await-in-loop */
+import JSZip from 'jszip';
+
+import alertCaller from 'app/actions/alert-caller';
+import deviceMaster from 'helpers/device-master';
+import dialog from 'implementations/dialog';
+import formatDuration from 'helpers/duration-formatter';
+import i18n from 'helpers/i18n';
+import progressCaller from 'app/actions/progress-caller';
+import storage from 'implementations/storage';
+
+export const downloadCameraData = async (deviceName: string): Promise<void> => {
+  const tBackup = i18n.lang.camera_data_backup;
+  const progressId = 'camera-data-backup';
+  progressCaller.openSteppingProgress({
+    id: progressId,
+    message: tBackup.checking_pictures,
+  });
+  const zip = new JSZip();
+  try {
+    let canceled = false;
+    const downloadFiles = async (names: string[], dirName: string) => {
+      if (canceled) return;
+      const s = Date.now();
+      zip.folder(dirName);
+      for (let i = 0; i < Math.min(names.length, 10); i += 1) {
+        const fileName = names[i];
+        const res = await deviceMaster.downloadFile(dirName, fileName, ({ left, size }) => {
+          const current = 1 - left / size;
+          const totalProgress = (current + i) / names.length;
+          const timeElapsed = (Date.now() - s) / 1000;
+          const timeLeft = formatDuration(timeElapsed / totalProgress - timeElapsed);
+          progressCaller.update(progressId, {
+            message: `${tBackup.downloading_data} ${dirName} ${i + 1}/${names.length}<br/>${
```

Comment:
Maybe add a variable `length = Math.min(names.length, 10)` and then generate `totalProgress` and progress `message` with it instead of `names.length`?